src/init.c

```diff
@@ -57,48 +58,21 @@ __attribute__((constructor)) void nm_init() {
     }
     #endif
 
-    NM_LOG("init: finding config files");
+    NM_LOG("init: updating config");
 
-    nm_config_file_t *files = nm_config_files(&err);
+    bool upd = nm_global_config_update(&err);
     if (err) {
-        NM_LOG("init: could not scan for config files: %s", err);
-        free(err);
-        return;
+        NM_LOG("init: error parsing config, will show a menu item with the error: %s", err);
     }
 
-    NM_LOG("init: parsing config");
-
-    size_t items_n;
-    nm_menu_item_t **items;
-    nm_config_t *cfg;
-
-    cfg = nm_config_parse(files, &err);
-    nm_config_files_free(files);
-
-    if (!cfg && err) {
-        NM_LOG("error: could not parse config: %s, creating error item in main menu instead", err);
-
-        items_n  = 1;
-        items    = calloc(items_n, sizeof(nm_menu_item_t*));
-        items[0] = calloc(1, sizeof(nm_menu_item_t));
-        items[0]->loc = NM_MENU_LOCATION_MAIN_MENU;
-        items[0]->lbl = strdup("Config Error");
-        items[0]->action = calloc(1, sizeof(nm_menu_action_t));
-        items[0]->action->arg = strdup(err);
-        items[0]->action->act = NM_ACTION(dbg_msg);
-        items[0]->action->on_failure = true;
-        items[0]->action->on_success = true;
-
-        free(err);
-    } else {
-        NM_LOG("init: generating items");
-        nm_config_generate(cfg);
-
-        NM_LOG("init: getting menu");
-        if (!(items = nm_config_get_menu(cfg, &items_n))) {
-            NM_LOG("error: could not allocate memory, stopping");
-            goto stop_fs;
-        }
+    size_t ntmp;
+    if (!upd) {
+        NM_LOG("init: no config file changes detected for initial config update (it should always return an error or update), stopping (this is a bug; err should have been returned instead)");
+        return;
+    } else if (nm_global_config_items(&ntmp)) {
+        NM_LOG("init: warning: no menu items returned by nm_global_config_items, ignoring for now (this is a bug; it should always have a menu item whether the default, an error, or the actual config)");
+    } else if (nm_global_config_items(&ntmp)) {
+        NM_LOG("init: warning: size returned by nm_global_config_items is 0, ignoring for now (this is a bug; it should always have a menu item whether the default, an error, or the actual config)");
     }
```

Comment:
*Scratches head* what are you doing here? Shouldn't the first if negate its test? And the second isn't checking the variable the log message says it is.

Reply:
Oh, I meant to check the variables... I'll fix that.